collection/lens.rkt

```diff
@@ -39,6 +67,21 @@
   (make-lens (λ (target) (nth target index))
              (λ (target value) (set-nth target index value))))
 
+(define (mapper-lens lens)
```

Comment:
But unlike `ref-lens` and `subsequence-lens`, `mapper-lens` takes a lens, not a procedure like it would if it was simply a partially applied version of `map`. But actually, because of that @jackfirth suggested, for consistency with other functions (`lens-compose`, `lens-zoom`) that consume lenses, that it could be named `lens-mapper`. I'm not sure which naming convention is best though.